c2rust-analyze/src/main.rs

```diff
@@ -457,6 +465,16 @@ fn body_owners_postorder(tcx: TyCtxt) -> Vec<LocalDefId> {
         if seen.contains(&root_ldid) {
             continue;
         }
+
+        match tcx.def_kind(root_ldid) {
+            DefKind::Fn => {}
+            DefKind::AnonConst => continue,
+            dk => panic!(
```

Comment:
A panic I notice when testing a `lighttpd` example is:

```
thread 'rustc' panicked at 'unexpected def_kind AssocFn for body_owner DefId(0:40 ~ lighttpd[c91b]::{impl#1}::clone)', c2rust-analyze/src/main.rs:472:19
```

so it seems like `AssocFn` should be handled here as well

Reply:
Interesting - I think this should be trivial to fix, but I'll add a test case using `#[derive(Clone)]` just to be sure.

Reply:
The trivial fix worked - the analysis no longer panics on seeing `impl Clone for ...` blocks.  However, field accesses on ADTs are still unsupported, so deriving `Clone` without `Copy` on a non-empty struct will cause a different panic.  (Deriving Clone and Copy together is fine, since in that case the generated Clone impl just does a copy rather than cloning field by field.)